### PR TITLE
Make Torch an optional dependency

### PR DESCRIPTION
**Pull request by suhacker1**
This PR makes PyTorch an optional dependency. It also updates the PyTorch file formats list both to clarify it and fix a typo.

### CODE REVIEW

**Comment by suhacker1** [comment]
Since PyTorch is now an optional dependency, we should update the installation instructions in the README when the new version is out.

File: fickling/pytorch.py [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
 import fickling.polyglot
 from fickling.fickle import Pickled
 
+try:
+    import torch
+except ModuleNotFoundError:
+    raise ImportError(
+        "The 'torch' module is required for this functionality."
+        "PyTorch is now an optional dependency in Fickling."
+        "Please use pip install '.[torch]'"
```
Comment (Boyan-MILANOV) [inline fickling/pytorch.py:16]: Shouldn't this be `pip install fickling[torch]`? I'm assuming people won't install by cloning the repo but do it from PyPI directly so `.[torch]` doesn't apply to them.

File: README.md
Comment (Boyan-MILANOV) [inline README.md:1]: As per your comment in the PR discussions, I think you should update the README with the new install instructions in this PR already. They go together.

If we don't want to have this pushed to `master` before a new release is out then it's perhaps time to have a `dev` working branch and use `master` only for releases. In fact I think we should do it anyway if we keep developing new `fickling` features

**Review by Boyan-MILANOV — CHANGES_REQUESTED** [review]

File: README.md
Comment (suhacker1) [reply README.md:1]: That makes sense. I'll add this and merge this PR right before making a new release then to make this easier in this case. But we can discuss the `dev` branch later.

**Review by Boyan-MILANOV — APPROVED** [review]
LGTM!

**Comment by drdavella** [comment]
Hi! We appreciate the work you're doing and especially this change to make Torch and optional dependency. I was just curious whether there are any plans to release this change in the near future? Thank you!

**Comment by suhacker1** [comment]
@drdavella We published the new release! Please don't hesitate to reach out with any issues or feature requests!

**Pull request merged**